### DISCUSSION TITLE
Validation in action contact()

### DISCUSSION THREAD

**Discussion by ondics** [yiisoft/demo#427] [Q&A]
I try to understand where the `$validator` object comes from in `src/Contact/ContactController.php`

```
use Yiisoft\Validator\ValidatorInterface;
...
final class ContactController
{
   ...
   public function contact(
       ValidatorInterface $validator,
       ServerRequestInterface $request
   ): ResponseInterface {

}

```

The action `contact` is defined in `./config/routes.php`:

```
return [
    // Lonely pages of site
    Route::get('/')
        ->action([SiteController::class, 'index'])
        ->name('site/index'),
    Route::methods([Method::GET, Method::POST], '/contact')
        ->action([ContactController::class, 'contact'])
        ->name('site/contact'),
...
```

But there is no link to any `Validator` defined nor any injection of an ValidatorInterface or a model class.

**Comment by samdark** [comment]
Yes, it's a new concept if you haven't used DI container in Yii2 (there it was optional). Here you go: https://github.com/yiisoft/docs/blob/master/guide/en/concept/di-container.md It's a draft documentation but, I think, explains it well.

**Reply by ondics** [reply]
Ok, thanks. Seems a bit challenging for newcomers to understand. If we want to win thousands of devs for Yii3, this needs a litte bit more elaborated.

Maybe the application structure should be explained here starting from `composer.json` settings in `extra/config-plugin`/...`:

```
"extra": {
        "branch-alias": {
            "dev-master": "3.0.x-dev"
        },
        "config-plugin-options": {
            "source-directory": "config"
        },
        "config-plugin": {
            ...
        }
```

**Reply by samdark** [reply]
Yes. Will be covered in the guide.
